src/test/java/book/store/onlinebookstore/book/BookControllerTest.java

```diff
@@ -0,0 +1,266 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"USER", "ADMIN"})
+    @DisplayName("Check creating book")
+    @Sql(scripts = "classpath:database.scripts/book/clear-book-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @SneakyThrows
+    void createBook_ValidCreateRequestDto_ReturnsBookDto() {
+        //given
+        var requestDto = new CreateBookRequestDto(
+                "Test book",
+                "Test author",
+                "123-123-0001",
+                BigDecimal.valueOf(100),
+                Set.of(1L, 2L),
+                null,
+                null
+        );
+
+        BookDto expected = new BookDto();
+        expected.setId(1L);
+        expected.setTitle(requestDto.title());
+        expected.setAuthor(requestDto.author());
+        expected.setIsbn(requestDto.isbn());
+        expected.setPrice(requestDto.price());
+        expected.setCategoriesIds(requestDto.categoriesIds());
+
+        //when
+        MvcResult result = mockMvc.perform(post("/api/books")
+                        .content(objectMapper.writeValueAsString(requestDto))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        //then
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+        Assertions.assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @DisplayName("Check if get all books returns list of books dto")
+    @WithMockUser
+    @Sql(scripts = "classpath:database.scripts/book/add-three-books.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database.scripts/book/clear-book-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @SneakyThrows
+    void getAll_ValidPageable_ReturnsListOfBooks() {
+        //given
+        BookDto book1 = new BookDto();
+        book1.setId(1L);
+        book1.setTitle("test-book1");
+        book1.setAuthor("test-author");
+        book1.setIsbn("123-123-0001");
+        book1.setPrice(BigDecimal.valueOf(100.99));
+        book1.setCategoriesIds(Set.of());

```

Comment:
i prefer use private method to get default value of test entity and just call him in test